File: lib/src/main/java/com/qianfanyun/lib/base/BaseImageLoaderStrategy.java
package com.qianfanyun.lib.base;

import android.net.Uri;
import android.widget.ImageView;

import com.qianfanyun.lib.ImageOptions;

import java.io.File;

/**
 * 图片加载能力的抽象接口
 *
 * @author ArcherYc
 * @date on 2019/2/28  3:46 PM
 * @mail 247067345@qq.com
 */
public interface BaseImageLoaderStrategy {

    void loadImage(ImageView imageView, String url);

    void loadImage(ImageView imageView, String url, ImageOptions options);

    void loadImage(ImageView imageView, int resId);

    void loadImage(ImageView imageView, int resId, ImageOptions options);

    void loadImage(ImageView imageView, File file);

    void loadImage(ImageView imageView, File file, ImageOptions options);

    void loadImage(ImageView imageView, Uri uri);

    void loadImage(ImageView imageView, Uri uri, ImageOptions options);
}

File: qfimage/src/main/java/com/qianfan/qfimage/base/BaseImageLoaderStrategy.kt
package com.qianfan.qfimage.base

import android.net.Uri
import android.widget.ImageView
import com.qianfan.qfimage.ImageOptions
import java.io.File

interface BaseImageLoaderStrategy {
    fun loadImage(imageView: ImageView, url: String)
    fun loadImage(imageView: ImageView, url: String, options: ImageOptions)
    fun loadImage(imageView: ImageView, resId: Int)
    fun loadImage(imageView: ImageView, resId: Int, options: ImageOptions)
    fun loadImage(imageView: ImageView, file: File)
    fun loadImage(imageView: ImageView, file: File, options: ImageOptions)
    fun loadImage(imageView: ImageView, uri: Uri)
    fun loadImage(imageView: ImageView, uri: Uri, options: ImageOptions)
    fun setDiskCacheSize(cacheSize: Long)
}